This patch is derived from the PyCA Cryptography 1.7.1 source, which has
the following copyright notice:

Copyright 2013-2016 The cryptography developers

The cryptography-1.7.1 source distribution states:

[These files are] dual licensed under the terms of the Apache License,
Version 2.0, and the BSD License. See the LICENSE file in the root of [the
cryptography-1.7.1 source distribution] for complete details.

Index: cryptography-1.7.1/src/_cffi_src/openssl/asn1.py
===================================================================
--- cryptography-1.7.1.orig/src/_cffi_src/openssl/asn1.py
+++ cryptography-1.7.1/src/_cffi_src/openssl/asn1.py
@@ -77,8 +77,6 @@ int ASN1_INTEGER_set(ASN1_INTEGER *, lon
 /*  ASN1 TIME */
 ASN1_TIME *ASN1_TIME_new(void);
 void ASN1_TIME_free(ASN1_TIME *);
-ASN1_GENERALIZEDTIME *ASN1_TIME_to_generalizedtime(ASN1_TIME *,
-                                                   ASN1_GENERALIZEDTIME **);
 ASN1_TIME *ASN1_TIME_set(ASN1_TIME *, time_t);
 
 /*  ASN1 UTCTIME */
@@ -152,6 +150,10 @@ int ASN1_STRING_set_default_mask_asc(cha
 
 int i2d_ASN1_TYPE(ASN1_TYPE *, unsigned char **);
 ASN1_TYPE *d2i_ASN1_TYPE(ASN1_TYPE **, const unsigned char **, long);
+
+/* ASN1_TIME * became const at some point */
+ASN1_GENERALIZEDTIME *ASN1_TIME_to_generalizedtime(ASN1_TIME *,
+                                                   ASN1_GENERALIZEDTIME **);
 """
 
 CUSTOMIZATIONS = """
Index: cryptography-1.7.1/src/_cffi_src/openssl/bio.py
===================================================================
--- cryptography-1.7.1.orig/src/_cffi_src/openssl/bio.py
+++ cryptography-1.7.1/src/_cffi_src/openssl/bio.py
@@ -10,7 +10,6 @@ INCLUDES = """
 
 TYPES = """
 typedef struct bio_st BIO;
-typedef void bio_info_cb(BIO *, int, const char *, int, long, long);
 typedef ... bio_st;
 typedef ... BIO_METHOD;
 typedef ... BUF_MEM;
@@ -69,11 +68,6 @@ BIO *BIO_new_fp(FILE *, int);
 BIO *BIO_new_fd(int, int);
 BIO *BIO_new_socket(int, int);
 long BIO_ctrl(BIO *, int, long, void *);
-long BIO_callback_ctrl(
-    BIO *,
-    int,
-    void (*)(struct bio_st *, int, const char *, int, long, long)
-);
 long BIO_int_ctrl(BIO *, int, long, int);
 size_t BIO_ctrl_pending(BIO *);
 size_t BIO_ctrl_wpending(BIO *);
@@ -125,8 +119,6 @@ int BIO_set_close(BIO *,long);
 int BIO_get_close(BIO *);
 int BIO_pending(BIO *);
 int BIO_wpending(BIO *);
-int BIO_get_info_callback(BIO *, bio_info_cb **);
-int BIO_set_info_callback(BIO *, bio_info_cb *);
 long BIO_get_buffer_num_lines(BIO *);
 long BIO_set_read_buffer_size(BIO *, long);
 long BIO_set_write_buffer_size(BIO *, long);
Index: cryptography-1.7.1/src/_cffi_src/openssl/evp.py
===================================================================
--- cryptography-1.7.1.orig/src/_cffi_src/openssl/evp.py
+++ cryptography-1.7.1/src/_cffi_src/openssl/evp.py
@@ -67,7 +67,6 @@ int EVP_MD_size(const EVP_MD *);
 EVP_PKEY *EVP_PKEY_new(void);
 void EVP_PKEY_free(EVP_PKEY *);
 int EVP_PKEY_type(int);
-int EVP_PKEY_size(EVP_PKEY *);
 RSA *EVP_PKEY_get1_RSA(EVP_PKEY *);
 DSA *EVP_PKEY_get1_DSA(EVP_PKEY *);
 DH *EVP_PKEY_get1_DH(EVP_PKEY *);
@@ -168,6 +167,9 @@ int EVP_PKEY_CTX_set_signature_md(EVP_PK
 int EVP_PBE_scrypt(const char *, size_t, const unsigned char *, size_t,
                    uint64_t, uint64_t, uint64_t, uint64_t, unsigned char *,
                    size_t);
+
+/* EVP_PKEY * became const at some point */
+int EVP_PKEY_size(EVP_PKEY *);
 """
 
 CUSTOMIZATIONS = """
Index: cryptography-1.7.1/src/_cffi_src/openssl/ssl.py
===================================================================
--- cryptography-1.7.1.orig/src/_cffi_src/openssl/ssl.py
+++ cryptography-1.7.1/src/_cffi_src/openssl/ssl.py
@@ -194,7 +194,6 @@ int SSL_get_error(const SSL *, int);
 int SSL_do_handshake(SSL *);
 int SSL_shutdown(SSL *);
 int SSL_renegotiate(SSL *);
-int SSL_renegotiate_pending(SSL *);
 const char *SSL_get_cipher_list(const SSL *, int);
 Cryptography_STACK_OF_SSL_CIPHER *SSL_get_ciphers(const SSL *);
 
@@ -435,6 +434,9 @@ long SSL_CTX_sess_cb_hits(SSL_CTX *);
 long SSL_CTX_sess_misses(SSL_CTX *);
 long SSL_CTX_sess_timeouts(SSL_CTX *);
 long SSL_CTX_sess_cache_full(SSL_CTX *);
+
+/* SSL * became const at some point */
+int SSL_renegotiate_pending(SSL *);
 """
 
 CUSTOMIZATIONS = """
